adapter/internal/eventhub/subscription.go

```diff
@@ -211,60 +225,103 @@ func InvokeService(endpoint string, responseType interface{}, queryParamMap map[
-				case *types.APIList:
-					apiListResponse := newResponse.(*types.APIList)
-					if logger.LoggerSubscription.Level == logrus.DebugLevel {
-						for _, api := range apiListResponse.List {
-							logger.LoggerSubscription.Infof("Received API List information for API : %s", api.UUID)
-						}
+		retrieveAPIList(response)
+	}
+}
+
+func retrieveAPIList(response response) {
+	responseType := reflect.TypeOf(response.Type).Elem()
+	newResponse := reflect.New(responseType).Interface()
+	if response.Error == nil && response.Payload != nil {
+		err := json.Unmarshal(response.Payload, &newResponse)
+		if err != nil {
+			logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the APIList response received for: "+
+				response.Endpoint, err)
+		} else {
+			switch t := newResponse.(type) {
+			case *types.APIList:
+				apiListResponse := newResponse.(*types.APIList)
+				if logger.LoggerSubscription.Level == logrus.DebugLevel {
+					for _, api := range apiListResponse.List {
+						logger.LoggerSubscription.Infof("Received API List information for API : %s", api.UUID)
 					}
-					if _, ok := APIListMap[response.GatewayLabel]; !ok {
-						// During the startup
-						APIListMap[response.GatewayLabel] = newResponse.(*types.APIList)
-					} else {
-						// API Details retrieved after startup contains single API per response.
-						if len(apiListResponse.List) == 1 {
-							APIListMap[response.GatewayLabel].List = append(APIListMap[response.GatewayLabel].List,
-								apiListResponse.List[0])
-						}
+				}
+				if _, ok := APIListMap[response.GatewayLabel]; !ok {
+					// During the startup
+					APIListMap[response.GatewayLabel] = newResponse.(*types.APIList)
+				} else {
+					// API Details retrieved after startup contains single API per response.
+					if len(apiListResponse.List) == 1 {
+						APIListMap[response.GatewayLabel].List = append(APIListMap[response.GatewayLabel].List,
+							apiListResponse.List[0])
 					}
-					xds.UpdateEnforcerAPIList(response.GatewayLabel, xds.MarshalAPIList(APIListMap[response.GatewayLabel]))
-				default:
-					logger.LoggerSubscription.Warnf("APIList Type DTO is not recieved. Unknown type %T", t)
 				}
+				xds.UpdateEnforcerAPIList(response.GatewayLabel, xds.MarshalAPIList(APIListMap[response.GatewayLabel]))
+			default:
+				logger.LoggerSubscription.Warnf("APIList Type DTO is not recieved. Unknown type %T", t)
 			}
 		}
 	}
 }
+
+func retrieveSubscriptionDataFromChannel(response response) {
+	responseType := reflect.TypeOf(response.Type).Elem()
+	newResponse := reflect.New(responseType).Interface()
+	err := json.Unmarshal(response.Payload, &newResponse)
+
+	if err != nil {
+		logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the response received for: "+response.Endpoint, err)
+	} else {
+		switch t := newResponse.(type) {
+		case *types.SubscriptionList:
+			logger.LoggerSubscription.Debug("Received Subscription information.")
```

Comment:
Shall we change these to be info logs. Along with size may be? 
@Rajith90 wdyt?

Reply:
+1

Reply:
done

Reply:
@Arshardh